clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5406,6 +5408,101 @@ ExprResult Sema::ConvertVectorExpr(Expr *E, TypeSourceInfo *TInfo,
                                    RParenLoc, CurFPFeatureOverrides());
 }
 
+ExprResult Sema::BuiltinInvoke(CallExpr *TheCall) {
+  auto Loc = TheCall->getBeginLoc();
+  auto Args = MutableArrayRef(TheCall->getArgs(), TheCall->getNumArgs());
+  assert(llvm::none_of(Args,
```

Comment:
Is this enforced earlier?  Why can't type dependent args make it here?

Reply:
I'm not sure whether dependent arguments can make it here, but I haven't been able to get any here. Should I remove the assertion?

Reply:
```
template <class T>
auto test(T v) {
  return __builtin_invoke(v);
}
```
should do the trick, right? Assuming I've set things up correctly on my end, this does not trigger the assert here.

Reply:
Yep, I would expect that to be enough to get it.  Looks like we must handle it elsewhere for these, I wasn't sure!